src/planner/ompl/CRRT.cpp

```diff
@@ -358,6 +362,18 @@ CRRT::Motion* CRRT::constrainedExtend(
       }
     }
 
+    // NOTE: (sniyaz) This check makes sure the resolution of the path on the
+    // constraint manifold and the resolution used for interpolation (*before*
+    // projection) do not differ wildly.
+    // TODO: Should we expose the factor to scale `mMaxStepsize` by?
+    double manifoldResolution = si_->distance(xstate, cmotion->state);
+    if (manifoldResolution > kLambda * mMaxStepsize)
+    {
+      // The resolution on the manifold is violated, so treat the projection as
+      // if it failed. This emulates the original CBiRRT paper correctly.
```

Comment:
1. Why have a `kLambda` at all? Why not just compare with `mMaxStepsize`?
2. Resolution of the path and resolution for interpolation are confusing terms in this context. Replace with: "Ensure that the random state projected onto the constraint manifold is not more than `mMaxStepsize` distance away from the nearest neighbor on the tree."

Reply:
I have not seen the original paper talk about this particular case actually. Could you point me to it? If not, best remove the reference to the paper and just explain the reason to add this block of code here.

Reply:
Nice catch on the original CBiRRT paper: I just re-read it and I don't think this is mentioned anywhere, so I must have been confused. I talked with Sidd a *long* time ago when I was writing my own CBiRRT code, and he mentioned that this is an important thing to do. I'll update the comment though, since it's wrong right now 😛 

The usage of `kLambda` is inspired by how OMPL's constrained statespaces handle this problem: their logic is that you want some "slack" between the resolution used during interpolation and the final resolution of the path on the constraint manifold. If you set `kLambda = 1`, any projection that increases the distance from the NN (even a little bit) is rejected, and that's probably too strict.

LMK your thoughts!

Reply:
Sure, a slack factor sounds good. Let's expose it, and change the name to denote what it implies. Maybe `mConstrainedExtensionMaxStepsize`?

Reply:
Done!